packages/workspace/src/core/project-graph/build-nodes/workspace-projects.ts

```diff
@@ -5,7 +5,7 @@ export function buildWorkspaceProjectNodes(
   addNode: AddProjectNode,
   fileRead: (s: string) => string
 ) {
-  const workspaceJsonProjects = Object.keys(ctx.workspaceJson.projects);
+  const workspaceJsonProjects = Object.keys(ctx.fileMap);
```

Comment:
Not in the case where fileMap is cached and incremental (some projects can be skipped). By doing this we can avoid the `mergeData` nonsense that was there before since no need to account for missing data.